packages/eslint-plugin/src/rules/prefer-find.ts

```diff
@@ -0,0 +1,283 @@
+import type { TSESLint, TSESTree } from '@typescript-eslint/utils';
+import { AST_NODE_TYPES } from '@typescript-eslint/utils';
+import { getScope, getSourceCode } from '@typescript-eslint/utils/eslint-utils';
+import type { RuleFix, SourceCode } from '@typescript-eslint/utils/ts-eslint';
+import * as tsutils from 'ts-api-utils';
+import type { Type } from 'typescript';
+
+import {
+  createRule,
+  getConstrainedTypeAtLocation,
+  getParserServices,
+  getStaticValue,
+  nullThrows,
+} from '../util';
+
+export default createRule({
+  name: 'prefer-find',
+  meta: {
+    docs: {
+      description:
+        'Enforce the use of Array.prototype.find() over Array.prototype.filter() followed by [0] when looking for a single result',
+      requiresTypeChecking: true,
+    },
+    messages: {
+      preferFind: 'Use .find(...) instead of .filter(...)[0]',
```

Comment:
[Phrasing] _"Use"_ is an action verb, which we'd normally put in the suggestion message. _"Prefer"_ is what we tend to put for the actual rule error complaint.

```suggestion
      preferFind: 'Prefer .find(...) instead of .filter(...)[0]',
```

Related: https://github.com/eslint-community/eslint-plugin-eslint-plugin/issues/368

Reply:
changed